src/core/client/admin/components/UserRole/UserRoleChangeContainer.tsx

```diff
@@ -80,22 +137,52 @@ const UserRoleChangeContainer: FunctionComponent<Props> = ({
   return (
     <UserRoleChange
       username={user.username}
-      onChangeRole={handleOnChangeRole}
-      onChangeModerationScopes={handleOnChangeModerationScopes}
-      role={user.role}
+      onChange={onChange}
+      onPromote={onPromote}
+      viewerRole={convertRole(viewer.role)}
+      role={convertRole(user.role)}
       scoped={user.moderationScopes?.scoped}
       moderationScopes={user.moderationScopes}
       moderationScopesEnabled={moderationScopesEnabled}
       query={query}
+      viewerScoped={
+        viewer.moderationScopes ? viewer.moderationScopes?.scoped : false
+      }
+      viewerSites={
+        viewer.moderationScopes
+          ? viewer.moderationScopes.sites?.map((s) => s.id)
+          : null
+      }
     />
   );
 };
 
+function convertRole(role: GQLUSER_ROLE_RL): GQLUSER_ROLE {
+  switch (role) {
+    case GQLUSER_ROLE.ADMIN:
+      return GQLUSER_ROLE.ADMIN;
+    case GQLUSER_ROLE.MODERATOR:
+      return GQLUSER_ROLE.MODERATOR;
+    case GQLUSER_ROLE.STAFF:
+      return GQLUSER_ROLE.STAFF;
+    case GQLUSER_ROLE.COMMENTER:
+      return GQLUSER_ROLE.COMMENTER;
+    default:
+      return GQLUSER_ROLE.COMMENTER;
+  }
+}
```

Comment:
Using the `GQLUSER_ROLE_RL` is important here, and should not be converted to a `GQLUSER_ROLE`. When referring to the _value_, you should always use `GQLUSER_ROLE` (Like when you're saying `viewer.role !== GQLUSER_ROLE.ADMIN`). When _typeing_ it, you should always use `GQLUSER_ROLE_RL`, as it ensures the type system will be forwards compatible with new role types.